tests/conftests/platforms.py

```diff
@@ -0,0 +1,166 @@
+"""
+In this we keep the platforms specify test configuration.
+The file is structured as follows:
+    - platform
+    - non_platform
+    - platform set
+    - non_platform set
+"""
+
+import pytest
+
+##########
+# Platform
+##########
+
+@pytest.fixture
+def ali(platform):
+    if platform != 'ali':
+        pytest.skip('test only supported on ali platform')
+
+@pytest.fixture
+def aws(platform):
+    if platform != 'aws':
+        pytest.skip('test only supported on aws platform')
+
+@pytest.fixture
+def azure(platform):
+    if platform != 'azure':
+        pytest.skip('test only supported on azure platform')
+
+@pytest.fixture
+def chroot(platform):
+    if platform != 'chroot':
+        pytest.skip('test only supported on chroot platform')
+
+@pytest.fixture
+def container(platform):
+    """
+    This fixture is an alias of "chroot" but does not use the "chroot" env.  However, it only needs
+    the underlying container for its tests.
+    """
+    if platform != 'container' or platform != 'chroot':
+        pytest.skip('test only supported on container platform')
+
+@pytest.fixture
+def firecracker(platform):
+    if platform != 'firecracker':
+        pytest.skip('test only supported on firecracker platform')
+
+@pytest.fixture
+def gcp(platform):
+    if platform != 'gcp':
+        pytest.skip('test only supported on gcp platform')
+
+@pytest.fixture
+def kvm(platform):
+    if platform != 'kvm':
+        pytest.skip('test only supported on kvm platform')
+
+@pytest.fixture
+def local(platform):
+    if platform != 'local':
+        pytest.skip('test only supported on local platform')
+
+@pytest.fixture
+def metal(platform):
+    if platform != 'metal':
+        pytest.skip('test only supported on metal platform')
+
+@pytest.fixture
+def openstack(platform):
+    if platform != 'openstack-ccee':
+        pytest.skip('test only supported on openstack')
+
+####################
+# Platform blacklist
+####################
+
+@pytest.fixture
+def non_ali(platform):
+    if platform == 'ali':
+        pytest.skip('test not supported on ali platform')
+
+@pytest.fixture
+def non_aws(platform):
+    if platform == 'aws':
+        pytest.skip('test not supported on aws platform')
+
+@pytest.fixture
+def non_azure(platform):
+    if platform == 'azure':
+        pytest.skip('test not supported on azure')
+
+@pytest.fixture
+def non_chroot(platform):
+    if platform == 'chroot':
+        pytest.skip('test not supported on chroot')
+
+@pytest.fixture
+def non_container(testconfig):
```

Comment:
In contrast, this is in fact referring to images having the feature [container](https://github.com/gardenlinux/gardenlinux/tree/main/features/container) enabled. 

This is not the opposite of what the current implementation of the `container` fixture above does.